src/Services/WordWrapper.vala

```diff
@@ -0,0 +1,89 @@
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+* Boston, MA 02111-1307, USA.
+*
+* Authored by: Natan Streppel <streppels at gmail.com>
+*/
+
+/**
+ * This class handles the wrapping logic for a given text within the bounds of 
+ * two informed strings, which act as "string wrappers" for the text 
+ */
+public class WordWrapper : Object {
+    private static Regex starting_whitespaces_regex;
+    private static Regex ending_whitespaces_regex;
+
+    static construct {
+        //  TODO: create 1 single regex to capture both starting and ending whitespaces only
+        WordWrapper.starting_whitespaces_regex = new Regex ("^(\\s)+", RegexCompileFlags.EXTENDED);
+        WordWrapper.ending_whitespaces_regex = new Regex ("(\\s)+$", RegexCompileFlags.EXTENDED);
+    }
+
+    /**
+     * Wraps this instance's text with first and second halves.
+     * Unwraps this instance's text element if already wrapped with first and second halves.
+     */
+     public static string apply_wrap (string original_text, string first_half, string second_half) {
+        string leading_spaces = "";
+        string trailing_spaces = "";
+
+        save_surrouding_spaces (original_text, ref leading_spaces, ref trailing_spaces);
+
+        string text = original_text.strip ();
+
+        if (already_wrapped (text, first_half, second_half)) {
+            // removes first and second halves within selected text
+            int head = first_half.char_count ();
+            int tail = text.char_count () - second_half.char_count ();
+            return get_return_string (text.substring (head, tail - head), leading_spaces, trailing_spaces);
+        } else {
+            // adds first and second halveschar_count ()
+            return get_return_string (first_half + text + second_half, leading_spaces, trailing_spaces);
+        }
+    }
+
+    /**
+     * Returns the given stripped string with its leading and trailing whitespaces back on
+     */
+    private static string get_return_string (string text, string leading_spaces, string trailing_spaces) {
+        return leading_spaces.concat(text).concat(trailing_spaces);
+    }
+
+    /**
+     * Counts the number of whitespace characters to the left and right of a given text
+     */
+    private static void save_surrouding_spaces (string text, ref string leading_spaces, ref string trailing_spaces) {
+        MatchInfo match_info;
+        // leading whitespaces
+        WordWrapper.starting_whitespaces_regex.match (text, 0, out match_info);
+        if (match_info.matches ()) {
+            leading_spaces = match_info.fetch (0);
+        }
+
+        // trailing whitespaces
+        WordWrapper.ending_whitespaces_regex.match (text, 0, out match_info);
+        if (match_info.matches ()) {
+            trailing_spaces = match_info.fetch (0);
+        }
+        print(@"leading spaces: \"$leading_spaces\" / trailing: \"$trailing_spaces\"\n");
```

Comment:
Don't leave the print :) 

If you want, you can use `debug ()` instead and call the app with `-d`

Reply:
And no need for the /n

Reply:
opsie, got me there. :sweat_smile: